whisperv6/mailserver_response.go

```diff
@@ -75,6 +76,7 @@ func tryCreateMailServerRequestFailedEvent(payload []byte) (*EnvelopeEvent, erro
 	errorMsg = string(remainder)
 
 	event := EnvelopeEvent{
+		Peer:  peer,
 		Hash:  requestID,
 		Event: EventMailServerRequestCompleted,
```

Comment:
Don't we want to change this? The method is named `tryCreateMailServerRequestFailedEvent` and here `EventMailServerRequestCompleted` is used. `EventMailServerRequestExpired` is never used except your most recent changes.

Reply:
I am looking at it a bit more and this might be actually correct code... 

```go
		Data: &MailServerResponse{
 			Error: errors.New(errorMsg),
 		},
```
vs 
```
		Data: &MailServerResponse{
 			LastEnvelopeHash: lastEnvelopeHash,
 			Cursor:           cursor,
 		},
```
both valid for `EventMailServerRequestCompleted`, however, the method names are very confusing.

Let's leave it as it is and I will create a separate ticket to change this.

Reply:
yeah, code seemed correct to me too. i thought your point was that it is confusing